src/components/clusterWizard/ClusterDetails.tsx

```diff
@@ -97,12 +97,15 @@ const getValidationSchema = (cluster?: Cluster) => {
   });
 };
 
-const validateClusterName = async (newName: string, existingClusterName?: string) => {
+// For performance reasons, this is validated on submit only
+const validateClusterName = async (newFullName: string, existingClusterFullName?: string) => {
   try {
     const { data: clusters } = await getClusters();
-    const names = clusters.map((c) => c.name).filter((n) => n !== existingClusterName);
-    if (names.includes(newName)) {
-      return `Name "${newName}" is already taken.`;
+    const names = clusters
+      .map((c) => `${c.name}.${c.baseDnsDomain}`)
+      .filter((n) => n !== existingClusterFullName);
+    if (names.includes(newFullName)) {
+      return `Name "${newFullName}" is already taken.`;
```

Comment:
Maybe this is clearer?
```suggestion
      return `"${newFullName}" name and domain combination is already taken.`;
```